go-controller/pkg/ovn/multipolicy_test.go

```diff
@@ -0,0 +1,487 @@
+				knetPod.Labels = podLabels
+			}
+			addPodNetwork(knetPod, testPod.secondaryPodInfos)
+			podsList = append(podsList, *knetPod)
+		}
+		fakeOvn.startWithDBSetup(dbSetup,
+			&v1.NamespaceList{
+				Items: namespaces,
+			},
+			&v1.PodList{
+				Items: podsList,
+			},
+			&knet.NetworkPolicyList{
+				Items: networkPolicies,
+			},
+			&mnpapi.MultiNetworkPolicyList{
+				Items: multinetworkPolicies,
+			},
+			&nettypes.NetworkAttachmentDefinitionList{
+				Items: nads,
+			},
+		)
+		for _, testPod := range pods {
+			testPod.populateLogicalSwitchCache(fakeOvn, getLogicalSwitchUUID(fakeOvn.controller.nbClient, nodeName))
+		}
+		var err error
+		if namespaces != nil {
+			err = fakeOvn.controller.WatchNamespaces()
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+		if pods != nil {
+			err = fakeOvn.controller.WatchPods()
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+		err = fakeOvn.controller.WatchNetworkPolicy()
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		for _, ocInfo := range fakeOvn.secondaryControllers {
+			for _, testPod := range pods {
+				testPod.populateSecondaryNetworkLogicalSwitchCache(fakeOvn, ocInfo)
+			}
+			if namespaces != nil {
+				err = ocInfo.bnc.WatchNamespaces()
+			}
+			if pods != nil {
+				err = ocInfo.bnc.WatchPods()
+			}
+			err = ocInfo.bnc.WatchMultiNetworkPolicy()
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+	}
+
+	getUpdatedInitialDB := func(tPods []testPod) []libovsdb.TestData {
+		updatedSwitchAndPods := getExpectedDataPodsAndSwitches(tPods, []string{nodeName})
+		secondarySwitchAndPods := getExpectedDataPodsAndSwitchesForSecondaryNetwork(fakeOvn, tPods)
+		if len(secondarySwitchAndPods) != 0 {
+			updatedSwitchAndPods = append(updatedSwitchAndPods, secondarySwitchAndPods...)
+		}
+		return append(getHairpinningACLsV4AndPortGroup(), updatedSwitchAndPods...)
+	}
+
+	ginkgo.Context("during execution", func() {
+		ginkgo.It("correctly creating an multinetworkPolicy with a peer namespace label", func() {
+			app.Action = func(ctx *cli.Context) error {
+				namespace1 := *newNamespace(namespaceName1)
+				namespace2 := *newNamespace(namespaceName2)
+				policy := getMatchLabelsNetworkPolicy(netPolicyName1, namespace1.Name,
+					namespace2.Name, "", true, true)
+				policy.Annotations = map[string]string{PolicyForAnnotation: util.GetNADName(nad.Namespace, nad.Name)}
+				netconf, err := util.ParseNetConf(nad)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				netInfo := util.NewNetInfo(netconf)
+				mpolicy := convertNetPolicyToMultiNetPolicy(policy)
+				startOvn(initialDB, []v1.Namespace{namespace1, namespace2}, nil, nil,
+					[]nettypes.NetworkAttachmentDefinition{*nad}, nil, nil)
+
+				ocInfo, ok := fakeOvn.secondaryControllers["network1"]
+				gomega.Expect(ok).To(gomega.Equal(true))
+				asf := ocInfo.asf
+				gomega.Expect(asf).NotTo(gomega.Equal(nil))
+				gomega.Expect(asf.ControllerName).To(gomega.Equal("network1-network-controller"))
```

Comment:
maybe we can make it a part of `startOvn`, so it checks expected secondary controllers are created based on given nads?